File: leetcode_26/solution.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i = 0
        while i < len(nums):
            while i + 1 < len(nums) and nums[i + 1] == nums[i]:
                del nums[i]
            i += 1
        return len(nums)

        # if len(nums) < 2:
        #     return len(nums)
        # k = 1
        # for i in range(1, len(nums)):
        #     if nums[i] != nums[i - 1]:
        #         nums[k] = nums[i]
        #         k += 1
        # return k


if __name__ == '__main__':
    nums = [0, 0, 1, 1, 1, 2, 2, 3, 3, 4]
    print(Solution().removeDuplicates(nums))

File: leetcode_26/solution.go
package main

import (
	"fmt"
)

//func removeDuplicates(nums []int) int {
//	if len(nums) <= 1 {
//		return len(nums)
//	}
//
//	i, j := 0, 1
//	for j < len(nums) {
//		if nums[j] != nums[i] {
//			i, j = i + 1, j + 1
//		} else {
//			nums = append(nums[:i], nums[j:]...)
//		}
//	}
//
//	return len(nums)
//}

//func removeDuplicates(nums []int) int {
//	if len(nums) <= 1 {
//		return len(nums)
//	}
//
//	i := 0
//	for j := 1; j < len(nums); j++ {
//		if nums[j] != nums[i] {
//			nums[i + 1] = nums[j]
//			i++
//		}
//	}
//	return i + 1
//}

func removeDuplicates(nums []int) int {
	if len(nums) <= 1 {
		return len(nums)
	}

	k := 1
	for j := 1; j < len(nums); j++ {
		if nums[j] != nums[k - 1] {
			nums[k] = nums[j]
			k++
		}
	}
	return k
}


func main()  {
	nums := []int{0,0,1,1,1,2,2,3,3,4}
	fmt.Println(removeDuplicates(nums), nums)
}
